category_encoders/one_hot.py

```diff
@@ -193,8 +193,10 @@ def get_dummies(X_in, cols=None):
 
         bin_cols = []
         for col in cols:
-            classes = [x for x in set(X[col].values.tolist()) if np.isfinite(x)]
-            for class_ in classes:
+            col_tuples = [class_map['mapping'] for class_map in ordinal_encoder.mapping if class_map['col'] == col][0]
+            fit_classes = set(col_val[1] for col_val in col_tuples)
+            input_classes = set(x for x in set(X[col].values.tolist()) if np.isfinite(x))
+            for class_ in (fit_classes | input_classes):
```

Comment:
Yes, that makes sense. Otherwise there will be the same issues with different sizes in the sklearn pipeline. For a single encoder the output shape should always have the same number of columns, no matter the input. I'll add the change.